cmd/telegraf/versioninfo.json

```diff
@@ -0,0 +1,43 @@
+{
```

Comment:
This version info is not correct for master:

- Last release: `1.21.2`
- master: `1.22.0-9ff4330b`
- nightly: `1.22.0-10ef264f`
- Next release: `1.22.0`

As-is, this would not generate the correct version if a user were to run `make` or for the nightlies. I would greatly prefer we say that version info is determined via `--version`.

I lean toward not doing this at all or as I said above doing this only for releases and making it part of the release build process and not in-tree. This means nightlies and local builds would not get this data, but I think that is entirely fine.

Reply:
Good point, I added a program to generate the `versioninfo.json` instead so that the version info will match whatever the makefile will produce so the version will match the correct release, nightly and local version. Was this your only hesitation for keeping this info to only release builds or did you have another concern? I don't see any harm running `goversioninfo` for all windows binaries we create.

Reply:
I believe this resolves my comments about the versions not matching. I'd like to play with this more to see how it works across the various release types we have + see how it works with cross-builds (i.e. I want to build a windows binary on my mac m1 still).

Reply:
Sounds good to me, testing this more would be good. I tested building it on windows and the CI cross builds it on Linux which I've tried out as well, but I don't have access to a Mac so that would be good to try.